Nonprofit-Exchange-Hub/web-app
# 378/cleaner signup flow

## Dev Summary

This PR implements the changes proposed in #378 . The overall idea is to make the signup step be email verification, followed by the user profile form once the email has been verified. This PR adds the corresponding changes to the `account-manager-controller`. UI changes haven't yet been implemented.




## Test Plan


repro steps:
***Reproducing this change will require using swagger. Note that `/register` is now `/signup`***
1. Run the backend and open http://localhost:3001/docs#/auth/AccountManagerController_signup
2. Enter this as the body
```json
{
    "email": "peter.parker@example.com",
    "password": "Secret1234%",
    "firstName": "user21",
    "last_name": "last 21"
}
```
Expect a json response with the user record.

3. Login at http://localhost:3001/docs#/auth/AccountManagerController_login
```json
{
    "email": "peter.parker@example.com",
    "password": "Secret1234%"
}
```
Expect the user json record

4.  Go to the update endpoint http://localhost:3001/docs#/auth/AccountManagerController_update. Enter any data and expect the data to be updated.

The steps below are tested with e2e tests.



## Resources
